File: test-suite-graal-function/src/test/java/example/MockTenancyIdProvider.java
package example;

import io.micronaut.context.annotation.Context;
import io.micronaut.context.annotation.Replaces;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.oraclecloud.core.TenancyIdProvider;
import jakarta.inject.Singleton;

@Context
@Singleton
@Replaces(TenancyIdProvider.class)
public class MockTenancyIdProvider implements TenancyIdProvider {

    public static String tenancyId = "test-tenancyId";


    @Nullable
    @Override
    public String getTenancyId() {
        return tenancyId;
    }
}

File: docs-examples/example-function-kotlin/src/test/kotlin/example/mock/MockTenancyIdProvider.kt
package example.mock

import io.micronaut.context.annotation.Context
import io.micronaut.context.annotation.Replaces
import io.micronaut.core.annotation.Nullable
import io.micronaut.oraclecloud.core.TenancyIdProvider
import jakarta.inject.Singleton

@Context
@Singleton
@Replaces(TenancyIdProvider::class)
class MockTenancyIdProvider : TenancyIdProvider {
    override fun getTenancyId(): @Nullable String {
        return MockData.tenancyId
    }
}
